packages/@uppy/companion/src/server/helpers/youtube_dl.js

```diff
@@ -3,10 +3,10 @@ const youtubedl = require('youtube-dl-progress-improved')
 // Downloads can take a lonnnnnng time
 const TIMEOUT = 30 * 60 * 1000
 
-function streamFile (url, output) {
+function streamFile (url, isAudio, output) {
   return youtubedl.download(url, {
     output,
-    format: 'worstvideo[height >= 480][ext=mp4]+[ext=m4a]/mp4',
+    format: isAudio ? 'bestaudio[ext=m4a]/m4a' : 'worstvideo[height >= 480][ext=mp4]+[ext=m4a]/mp4',
```

Comment:
@zackbloom, I went w/`bestaudio` here since I figured filesize didn't matter in the same way it did for video; but, I don't know enough about yt-dlp options and maybe this is overkill.

Seems easy enough to update as needed, though :)

Reply:
I think thats totally fine, yea.

Reply:
Two questions I have:

-- Is there any value in adding `--audio-format mp3`? I really have no idea if it's important or not to get everything as mp3 (maybe there are some weird formats, or providers who don't actually provide m4a files).
-- Should we add `--audio-only`? Or does that not matter?

I'm thinking that those two options might be useful for when a provider has a combined audio/video file in a single container, and/or when they provide audio in like oog or whatever, not m4a.

Reply:
I originally tried an even simpler `mp3` format request and it failed because the test video didn't support that audio format. I don't think I tried many different ones, so there is a chance we run into the exact same problem (but different) for the `m4a` type? I suppose it is worth a little more trial and error here to see what would be the best idea to use.

Before using `m4a`, I also tried the other audio option `webm`; but, that caused issues because [it wasn't recognized as an audio type by Loom](https://github.com/SendHeirloom/Loom/blob/9b0f2995b248bd0b49641510cdfd4f282520be77/worker/tasks/details.go#L24).

I didn't realize there was a separate `--audio-only` flag; will make sure to include that in the things I try when I circle back around to this (likely next week).

Reply:
> maybe there are some weird formats, or providers who don't actually provide m4a files

I checked nearly 100 videos and all of them only had different `m4a` and `webm` options for audio only. After picking a few random ones, I cycled through two playlists trying to capture the best chance of varying audio types: one of classic (old) videos and another of high quality FLAC audio example videos.

---

I say we leave this part as-is for now and, if you ever notice an error in Sentry or have someone complain, we can look at the specific video's attributes.

Reply:
I'm still trying to figure out how to correctly include `--audio-only` since it seems like it wouldn't hurt to add; however, as I'm looking at `--list-formats` again / more closely, I'm realizing that there are only "audio only" versions of `m4a` files. But, there are both audio only and combined versions of `webm` (which explains why I was getting full videos when I picked that type, since I had noticed "audio only" versions, and why Loom doesn't consider it an audio extension).

Reply:
Okay, final note here, I don't think I need to do anything for `--audio-only` either; according to [the docs](https://github.com/yt-dlp/yt-dlp#format-selection) (emphasis _theirs_ :joy:), the `bestaudio` format includes `--audio-only` implicitly:

> `ba`, `bestaudio`: Select the best quality **audio-only** format.

Reply:
Oops, almost forgot to check non-YT videos for all of these. I tried an example on both Vimeo and FB and they each had `m4a` audio only options; in fact, that was the only audio only version on the FB video :+1:

Reply:
yea, I guess this can be tabled until / unless we find some provider who doesn't provide the file format.